invenio_rdm_records/ext.py

```diff
@@ -75,7 +80,31 @@ def verify_token():
                 # we need to add the LinkNeed here
                 if hasattr(g, "identity"):
                     g.identity.provides.add(LinkNeed(data["id"]))
-
+            else:
+                if not current_app.config.get("RDM_RESOURCE_ACCESS_TOKENS_ENABLED"):
+                    raise RATFeatureDisabledError()
+                # for records: "api/records/<pid_value>/files/<filename>/content"
+                # for drafts: "api/records/<pid_value>/draft/files/<key>/content"
+                filename = (
+                    request.view_args.get("filename")
+                    or request.view_args.get("key")
+                    or None
+                )
+                pid_value = request.view_args.get("pid_value", None)
```

Comment:
major: do we need to fetch and resolve these objects here? I would kind of "blindly" assume that the JWT payload includes the "claimed" subject and create a `RATNeed(**subject)`. Later in the permissions generator, these can be explicitly checked.

To translate into "human" terms, the flow would look something like:

- Client: *Hey, here's a RAT token, in which I claim that I can access as `user: 1234` on a record with `id: "m5ykg-ps142"`, the file with `key: "data.zip"`*
- Server: Ok... I verify that your token is indeed validly signed, so you can go on with your claims
- Client: *Noice*
- (...a few moments later in the permissions check of `get_content_files` of `/api/records/m5ykg-ps142/draft/files/data.zip`...)
- Server: Ok, from the permission policy it says I either need the **owner of the record**, an **admin role**, or someone with a RAT token that claims he can access as the owner of the record (`user: 1234`) this specific record (`id: "m5ykg-ps142"`) and file (`key: "data.zip"`)
- Client: *Oh boi, guess what I happen to have in my pocket* 😏
- Server: LGTM, here's your file contents 🫡